src/components/TemplateSourceFields/index.jsx

```diff
@@ -23,27 +49,58 @@ class TemplateSourceFields extends React.Component {
     setEmailTemplateSource(EMAIL_TEMPLATE_SOURCE_NEW_EMAIL);
   }
 
-  selectRenderField({ input, options }) {
+  selectRenderField({ input, templateOptions, changeFromEmailTemplate }) {
     return (
       <div className="template-select-wrapper mb-3">
         <label htmlFor="templateNameSelect">Template Name</label>
         <Input
           {...input}
           type="select"
           id="templateNameSelect"
-          options={options}
+          options={templateOptions}
+          onChange={changeFromEmailTemplate}
         />
       </div>
     );
   }
 
   updateState(emailTemplateSource) {
-    const { setEmailTemplateSource } = this.props;
+    const {
+      setEmailTemplateSource, allEmailTemplates,
+    } = this.props;
     setEmailTemplateSource(emailTemplateSource);
+    if (emailTemplateSource === EMAIL_TEMPLATE_SOURCE_FROM_TEMPLATE) {
+      this.makeOptions(this.props);
+      this.dispatchUpdatedTemplate(allEmailTemplates);
+    }
+  }
+
+  makeOptions(props) {
+    if (props.allEmailTemplates.length > 0) {
+      const templateOptions = [];
+      props.allEmailTemplates.forEach((emailTemplate) => {
+        if (emailTemplate.email_type === props.emailTemplateType) {
+          templateOptions.push({ value: emailTemplate.name, label: emailTemplate.name });
+        }
+      });
+      this.setState({ templateOptions });
+    }
+  }
+
+  dispatchUpdatedTemplate(emailTemplate) {
+    const { saveTemplateSuccess, emailTemplateType } = this.props;
+    if (emailTemplate.length > 0) {
+      saveTemplateSuccess(emailTemplateType, emailTemplate[0]);
```

Comment:
I think it is better to have separate action fired when an option is changed in dropdown something like `CURRENT_TEMPLATE_VISIBLE` and store the related state. `saveTemplateSuccess` is an entirely different action which is not related to dropdown option change.